src/features/smartSearch/hooks/useTitlesForCampaignParticipation.ts

```diff
@@ -0,0 +1,88 @@
+import { loadItemIfNecessary } from 'core/caching/cacheUtils';
+import {
+  activityLoad,
+  activityLoaded,
+  campaignLoad,
+  campaignLoaded,
+  locationLoad,
+  locationLoaded,
+} from '../store';
+import { useApiClient, useAppDispatch, useAppSelector } from 'core/hooks';
+import {
+  ZetkinActivity,
+  ZetkinCampaign,
+  ZetkinLocation,
+} from 'utils/types/zetkin';
+
+type CampaignParticipationTitles = {
+  activityTitle: string | null;
+  campaignTitle: string | null;
+  locationTitle: string | null;
+};
+
+export default function useTitlesForCampaignParticipation(
+  orgId: number,
+  activityId: number | undefined,
+  campaignId: number | undefined,
+  locationId: number | undefined
+): CampaignParticipationTitles {
+  const apiClient = useApiClient();
+  const dispatch = useAppDispatch();
+  const activityItem = useAppSelector((state) =>
+    state.smartSearch.activityList.items.find((item) => item.id == activityId)
+  );
+  const campaignItem = useAppSelector((state) =>
+    state.smartSearch.campaignList.items.find((item) => item.id == campaignId)
+  );
+  const locationItem = useAppSelector((state) =>
+    state.smartSearch.locationList.items.find((item) => item.id == locationId)
+  );
+
+  const titles: CampaignParticipationTitles = {
+    activityTitle: null,
+    campaignTitle: null,
+    locationTitle: null,
+  };
+
+  if (activityId !== undefined && activityId !== null) {
```

Comment:
I would liked to have more time to change this condition pattern, I'm curious to know how you are going to implement it :)

Reply:
This code is synonymous (in this context) with `if (!activityId)` so I would probably do that. But I haven't looked through the code yet so I don't know if I would make any more structural changes.